File: problems/002/002.lua
posix = require "posix"

local function f()
    local ans, curr, next_ = 0, 1, 2

    while curr < 4000000 do
        if curr % 2 == 0 then
            ans = ans + curr
        end

        local tmp = next_
        next_ = next_ + curr
        curr = tmp
    end

    return ans
end

function toNs(ts)
    return ts[1] * 1000000000 + ts[2]
end

if #arg == 0 then
    print(f())
elseif #arg == 1 then
    iters = tonumber(arg[1])
    start = { posix.clock_gettime("monotonic") }
    for i = 1,iters do
        f()
    end
    end_ = { posix.clock_gettime("monotonic") }
    print(toNs(end_) - toNs(start))
end

File: problems/002/002.js
function f() {
    var ans = 0;
    var curr = 1;
    var next = 2;
    var tmp;

    while (curr < 4000000) {
        if (curr % 2 === 0) {
            ans += curr;
        }

        tmp = next;
        next += curr;
        curr = tmp;
    }

    return ans;
}

function toNs(ts) {
    return ts.sec * 1000000000 + ts.nsec;
}

var clock = require('posix-clock');

if (process.argv.length === 3) {
    var i;
    var iters = parseInt(process.argv[2], 10);

    var start = clock.gettime(clock.MONOTONIC);
    for (i = 0; i < iters; i++) {
        f();
    }
    var end = clock.gettime(clock.MONOTONIC);

    console.log(toNs(end) - toNs(start));
} else {
    console.log(f());
}
